librubyfmt/src/lib.rs

```diff
@@ -0,0 +1,213 @@
+//#![deny(warnings, missing_copy_implementations)]
+use std::ffi::CString;
+#[macro_use]
+extern crate lazy_static;
+
+use std::io::{BufReader, Write, Cursor};
+use std::slice;
+
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
+pub type RawStatus = i64;
+
+mod breakable_entry;
+mod comment_block;
+mod de;
+mod delimiters;
+mod file_comments;
+mod format;
+mod intermediary;
+mod line_metadata;
+mod line_tokens;
+mod parser_state;
+mod render_queue_writer;
+mod ripper_tree_types;
+mod ruby;
+mod types;
+
+use file_comments::FileComments;
+use parser_state::ParserState;
+use ruby::VALUE;
+
+#[cfg(debug_assertions)]
+use log::debug;
+#[cfg(debug_assertions)]
+use simplelog::{CombinedLogger, Config, LevelFilter, TermLogger, TerminalMode};
+
+type RubyfmtResult<T = ()> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+extern "C" {
+    pub fn Init_ripper();
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct FormatBuffer {
+    pub bytes: *const libc::c_char,
+    pub count: i64,
+}
```

Comment:
What do you think about having this wrap `Box<str>` or `Box<CStr>`?

Reply:
why?

Reply:
Because you're reimplementing it anyway, why not do it without unsafe code